core/ArchiveProcessor.php

```diff
@@ -414,30 +403,40 @@ protected function getOperationForColumns($columns, $defaultOperation)
 
     protected function enrichWithUniqueVisitorsMetric(Row $row)
     {
-        // skip unique visitors metrics calculation if calculating for multiple sites is disabled
-        if (!$this->getParams()->isSingleSite()
-            && $this->skipUniqueVisitorsCalculationForMultipleSites
-        ) {
-            return;
-        }
-
         if ($row->getColumn('nb_uniq_visitors') === false
             && $row->getColumn('nb_users') === false
         ) {
             return;
         }
 
-        if (!SettingsPiwik::isUniqueVisitorsEnabled($this->getParams()->getPeriod()->getLabel())) {
+        $periodLabel = $this->getParams()->getPeriod()->getLabel();
+
+        if (!SettingsPiwik::isUniqueVisitorsEnabled($periodLabel)) {
             $row->deleteColumn('nb_uniq_visitors');
             $row->deleteColumn('nb_users');
             return;
         }
 
-        $metrics = array(
-            Metrics::INDEX_NB_USERS
-        );
+        $sites = $this->getIdSitesToComputeNbUniques();
+
+        if (count($sites) > 1 && Rules::shouldSkipUniqueVisitorsCalculationForMultipleSites()) {
+            if ($periodLabel != 'day') {
```

Comment:
when period = day, we keep the aggregated unique visitor metric which is calculated by summing the unique visitor metric of each indidividual website